### PR TITLE
1a convert salesforce ups consumer to the/ch132064

### PR DESCRIPTION
**Pull request by soobin-guild**
## Description of Changes
- Cloned event consumer template and made changes to configure repo as `sfdc-ups-consumer` 

## Link to Story
[1a convert salesforce ups consumer to the/ch132064]( https://app.clubhouse.io/guild/story/132064/1a-convert-salesforce-ups-consumer-to-the-new-pattern)

## Checklist
- [ ] Has tests.
- [ ] Documentation updated.
- [ ] Passes CI.
- [x] Makes author proud.

### CODE REVIEW

**Review by scodan — APPROVED** [review]

**Review by joemarkey-guild — APPROVED** [review]

File: infrastructure/config/default.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 # The github repo of this event bus consumer.
-githubRepo: https://github.com/GuildEducationInc/sfdc-event-bus-consumer-template
+githubRepo: https://github.com/GuildEducationInc/sfdc-ups-consumer
 
 # The prefix of the event bus kinesis stream output export name.  This value can be found in
 # AWS CloudFormation, within the Outputs tab of the event-bus-<region>-<env> stack.  The value
 # is the prefix of the "Export name" value.  For example, the "canary" stream has an Export name
 # in dev of "canary-stream-arn-dev", so the value to use is "canary-stream-arn"
-eventStreamPrefix: canary-stream-arn
+eventStreamPrefix: user-profile-service-event-stream-arn-dev
```
Comment (mmindenhall) [inline infrastructure/config/default.yaml:16]: This should not include the `-dev` suffix.  We add that in the code based on the value of the `STAGE` env var.

File: package.json [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
     "test": "NODE_ENV= jest"
   },
   "engines": {
-    "node": "10.16.3"
+    "node": "10.23.0"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/GuildEducationInc/~/ggit/sfdc-event-bus-consumer-template.git"
+    "url": "git+https://github.com/GuildEducationInc/~/ggit/sfdc-ups-consumer.git"
```
Comment (mmindenhall) [inline package.json:15]: The value here should be just the url of the repo:  https://github.com/GuildEducationInc/sfdc-ups-consumer

File: package.json [hunk 3]
```diff
@@ -21,9 +21,9 @@
   "author": "Guild Education",
   "license": "UNLICENSED",
   "bugs": {
-    "url": "https://github.com/GuildEducationInc/sfdc-catalog-service-consumer/issues"
+    "url": "https://github.com/GuildEducationInc/sfdc-ups-consumer/issues"
```
Comment (mmindenhall) [inline package.json:24]: We don't use github issues.  Can you paste a clubhouse URL here for the SF B2C project?

File: package.json [hunk 4]
```diff
@@ -34,13 +34,13 @@
...
   },
   "devDependencies": {
-    "@aws-cdk/core": "1.83.0",
-    "@aws-cdk/aws-secretsmanager": "1.83.0",
-    "@aws-cdk/aws-iam": "1.83.0",
-    "@aws-cdk/aws-ssm": "1.83.0",
-    "@aws-cdk/aws-sqs": "1.83.0",
-    "@aws-cdk/aws-lambda": "1.83.0",
-    "@aws-cdk/aws-ec2": "1.83.0",
+    "@aws-cdk/core": "1.93.0",
```
Comment (mmindenhall) [inline package.json:37]: Thanks for updating to the latest!

**Comment by mmindenhall** [comment]
@soobin-guild, in addition to the comments above, you need to replace the test events (of type `meep-event`) with sample events of type `user-verification-event`, and make sure that all of the tests are passing via `npm test`.  Once that's done, let's jump on a call together and get this deployed into the AWS dev environment (leaving disabled for now).

File: infrastructure/config/default.yaml (hunk 1 above)
Comment (soobin-guild) [reply infrastructure/config/default.yaml]: `-dev-` was removed from the file

File: package.json (hunk 3 above)
Comment (soobin-guild) [reply package.json]: URL updated to ` "url": "https://app.clubhouse.io/guild/project/269/salesforce-b2c"`

File: package.json (hunk 2 above)
Comment (soobin-guild) [reply package.json]: repo url updated to `"url": "https://github.com/GuildEducationInc/sfdc-ups-consumer"`

**Comment by scodan** [comment]
Soobin, we will probably need unit tests that are specific to the json event as well. I missed that in my prior review.

**Review by scodan — COMMENTED** [review]
Does this need to be changed/updated as well in deploy.yaml?
CDK_VERSION: 1.83.0

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
     branches-ignore:
       - '**'
-#    branches:
-#      - develop
-#      - staging
-#      - main
+    branches:
+      - develop
+      - staging
+      - main
```
Comment (soobin-guild) [inline .github/workflows/deploy.yml:10]: @mmindenhall Changes made for CICD pipeline

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -29,6 +29,13 @@ jobs:
             SF_PASSWORD=${{ secrets.SF_PASSWORD_PROD }}
             SF_TOKEN=${{ secrets.SF_TOKEN_PROD }}
             STAGE=prod
+          elif [ "$GITHUB_REF" = "refs/heads/staging" ]; then
+            AWS_ACCESS_KEY_ID=${{ secrets.DEVOPS_STAGING_AWS_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY=${{ secrets.DEVOPS_STAGING_AWS_SECRET_ACCESS_KEY  }}
+            SF_USERNAME=${{ secrets.SF_USERNAME_STAGING }}
+            SF_PASSWORD=${{ secrets.SF_PASSWORD_STAGING }}
+            SF_TOKEN=${{ secrets.SF_TOKEN_STAGING }}
+            STAGE=staging
```
Comment (soobin-guild) [inline .github/workflows/deploy.yml:38]: @mmindenhall Staging credentials added

**Review by mmindenhall — APPROVED** [review]

**Comment by mmindenhall** [comment]
I resolved the conflicts to get this back to ready to merge.  @soobin-guild, can we get this merged?  Once we do that, we can flip the parameters to enable this and disable the old consumer so you can finish testing.

**Pull request merged**